src/sentry/integrations/slack/utils/channel.py

```diff
@@ -173,3 +178,111 @@ def get_channel_id_with_timeout(
             return id_data
 
     return prefix, None, False
+
+
+def get_channel_id_with_timeout_new(
+    integration: Integration | RpcIntegration,
+    name: Optional[str],
+    timeout: int,
+) -> Tuple[str, Optional[str], bool]:
+    """
+    Fetches the internal slack id of a channel.
+    :param integration: The slack integration
+    :param name: The name of the channel
+    :param timeout: Our self-imposed time limit.
+    :return: a tuple of three values
+        1. prefix: string (`"#"` or `"@"`)
+        2. channel_id: string or `None`
+        3. timed_out: boolean (whether we hit our self-imposed time limit)
+    """
+
+    payload = {
+        "exclude_archived": False,
+        "exclude_members": True,
+        "types": "public_channel,private_channel",
+    }
+
+    time_to_quit = time.time() + timeout
+
+    client = SlackClient(integration_id=integration.id)
+    id_data: Optional[Tuple[str, Optional[str], bool]] = None
+    found_duplicate = False
+    prefix = ""
+    cursor = ""
+    try:  # Check for channel
+        channel_id = check_for_channel(client, name)
+    except ApiError as e:
+        if str(e) != "channel_not_found":
+            raise e
+        # Check if user
+        while True:
```

Comment:
@chloeho7 Nit: I would pull up the `cursor` initialization here since we only need it for the while loop.